app/actions/websocket/channel.ts

```diff
@@ -162,8 +166,15 @@ export async function handleDirectAddedEvent(serverUrl: string, msg: any) {
 
     const {database} = operator;
 
+    if (ephemeral_store.creatingChannel) {
+        return; // We probably don't need to handle this WS because we provoked it
+    }
     try {
         const {channel_id: channelId} = msg.broadcast;
+        const channel = await getChannelById(database, channelId);
+        if (channel) {
```

Comment:
This check may be still useful. In case the websocket event takes long to reach, and we already created the channel, we make sure sure not to repeat the process.